model_signing/serialization/itemized.py

```diff
@@ -0,0 +1,228 @@
+# Copyright 2024 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Model serializers that build an itemized manifest."""
+
+import concurrent.futures
+import pathlib
+from typing import Callable, Iterable
+from typing_extensions import override
+
+from model_signing.hashing import file
+from model_signing.manifest import manifest
+from model_signing.serialization import dfs
+from model_signing.serialization import serialization
+
+
+class FilesSerializer(serialization.Serializer):
+    """Model serializers that produces an itemized manifest, at file level.
+
+    Traverses the model directory and creates digests for every file found,
+    possibly in parallel.
+
+    Since the manifest lists each item individually, this will also enable
+    support for incremental updates (to be added later).
+    """
+
+    def __init__(
+        self,
+        hasher_factory: Callable[[pathlib.Path], file.FileHasher],
+        max_workers: int | None = None,
```

Comment:
How do you feel about using `Optional[int]`?
It avoids the duplicated `None` :)

Reply:
The internal coding style now recommends the `int | None` version as `Optional[int]` could be read as both an argument that is optional and an argument that can be None.